Sources/ComposableArchitecture/Reducer/Reducers/BindingReducer.swift

```diff
@@ -70,7 +70,7 @@ where State == ViewAction.State {
 
   @inlinable
   public func reduce(into state: inout State, action: Action) -> Effect<Action> {
-    guard let bindingAction = self.toViewAction(action).flatMap(\.binding)
+    guard let bindingAction = self.toViewAction(action).flatMap({ $0.binding })
```

Comment:
```suggestion
    // NB: Using a closure and not a `\.binding` key path literal to avoid a bug with archives:
    //     https://github.com/pointfreeco/swift-composable-architecture/pull/2641
    guard let bindingAction = self.toViewAction(action).flatMap({ $0.binding })
```